internal/cache/sharedcache.go

```diff
@@ -0,0 +1,145 @@
+// Copyright 2025 The argocd-agent Authors
+//
```

Comment:
As a general rule for me, if you're adding a mutable `var` at package scope, it's going to make me go :thinking: when I see it. `var sharedCache = &SharedCache{}` here (and the corresponding logic) is an example of that. AFAICT there are no other mutable package-level fields in the codebase.

Rather than creating a singleton, why not just maintain a reference to cache in...

Agent:
`type Agent struct {` in `agent/agent.go`
- See `redisProxyMsgHandler *redisProxyMsgHandler` for an example of redis cache object I use in agent case


Principal:
`type Server struct {` in `server/server.go` (where we store most other principal state)
or perhaps
`type Manager struct {` in `internal/argocd/cluster/manager.go`